server/routes/appointments/appointmentsController.ts

```diff
@@ -1294,11 +1334,12 @@ export default class AppointmentsController {
   private async getSupplierAssessmentAppointmentFromDraftOrService(
     req: Request,
     res: Response,
-    supplierAssessment: SupplierAssessment
+    supplierAssessment: SupplierAssessment,
+    referralId?: string
   ): Promise<InitialAssessmentAppointment | null> {
     const { draftBookingId, id } = req.params
     if (draftBookingId) {
-      const fetchResult = await this.fetchDraftAppointmentOrRenderMessage(req, res)
```

Comment:
pass referral id through to draft process, so that back link is populated correctly